Muvr/Entities/MRManagedExerciseSession.swift

```diff
@@ -1,198 +1,146 @@
 import Foundation
 import CoreData
 import MuvrKit
-import CoreLocation
 
-class MRManagedExerciseSession: NSManagedObject, MKClassificationHintSource {
-    private var currentClassificationHint: MKClassificationHint?
-    private var exerciseIdCounts: [MKExerciseId : Int] = [:]
-    /// The estimated exercises
-    var estimated: [MKClassifiedExercise] = []
-    /// The exercise plan
-    var plan = MKExercisePlan<MKExerciseId>()
-    /// The intended exercise type
-    var intendedType: MKExerciseType?
+class MRManagedExerciseSession: NSManagedObject {
+    typealias ExerciseRow = (MKExercise.Id, MKExerciseType, NSTimeInterval, NSTimeInterval, [MKExerciseLabel])
+    /// The number of exercise ids for next estimates
+    private var exerciseIdCounts: [MKExercise.Id : Int] = [:]
+    /// The exercise detail the user has explicitly started
+    /// Viz ``setClassificationHint(_:labels:)`` and ``clearClassificationHint``
+    private(set) internal var classificationHints: [MKClassificationHint] = []
+    /// The accumulated exercise rows
+    private(set) internal var exerciseWithLabels: [ExerciseRow] = []
+    
+    /// The set of estimated exercises, used when the session is in real-time mode
+    var estimated: [MKExerciseWithLabels] = []
     /// The weight predictor
     var weightPredictor: MKScalarPredictor!
+    /// The weight predictor
```

Comment:
Comment should say ``The duration predictor`` (or remove comments as we can easily guess with the variable names ``durationPredictor``)